Description: Fixes GCC8 FTBFS due to included copy of SMESH.
Author: Kurt Kremitzki <kurt@kwk.systems>
Last-Updated: 2018-08-10
--- a/src/3rdParty/salomesmesh/inc/SMDS_MeshElement.hxx
+++ b/src/3rdParty/salomesmesh/inc/SMDS_MeshElement.hxx
@@ -216,8 +216,8 @@
 // WARNING: this comparator makes impossible to store both nodes and elements in the same set
 // because there are nodes and elements with the same ID. Use TIDTypeCompare for such containers.
 struct TIDCompare {
-  bool operator () (const SMDS_MeshElement* e1, const SMDS_MeshElement* e2) const
-  { return e1->GetID() < e2->GetID(); }
+  template<typename T> bool operator () (const T* e1, const T* e2) const
+  { return static_cast<const SMDS_MeshElement*>(e1)->GetID() < static_cast<const SMDS_MeshElement*>(e2)->GetID(); }
 };
 
 #endif
